examples/controls_example/public/react_controls/data_controls/types.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CanClearSelections } from '@kbn/controls-plugin/public';
+import { DataViewField } from '@kbn/data-views-plugin/common';
+import { Filter } from '@kbn/es-query';
+import {
+  HasEditCapabilities,
+  PublishesDataViews,
+  PublishesFilters,
+  PublishesPanelTitle,
+} from '@kbn/presentation-publishing';
+import {
+  ControlFactory,
+  ControlStateManager,
+  DefaultControlApi,
+  DefaultControlState,
+} from '../types';
+
+export type DataControlApi = DefaultControlApi &
+  Omit<PublishesPanelTitle, 'hidePanelTitle'> & // control titles cannot be hidden
+  HasEditCapabilities &
+  CanClearSelections &
+  PublishesDataViews &
+  PublishesFilters & {
+    setOutputFilter: (filter: Filter | undefined) => void; // a control should only ever output a **single** filter
+  };
+
+export interface DataControlFactory<
+  State extends DefaultDataControlState = DefaultDataControlState,
+  Api extends DataControlApi = DataControlApi
+> extends ControlFactory<State, Api> {
+  isFieldCompatible: (field: DataViewField) => boolean;
+  CustomOptionsComponent?: React.FC<{
+    stateManager: ControlStateManager<State>;
+    setControlEditorValid: (valid: boolean) => void;
+  }>;
+}
+
+export const isDataControlFactory = (factory: unknown): factory is DataControlFactory => {
```

Comment:
Maybe instead of taking `unknown`, this type guard should take a control factory type? Theoretically, I could put `{ isFieldCompatible: () => false }` in here and get back `true`.

Reply:
Good catch - done in 9449347830342338953e231e00753a87b3cdd633